lib/infra/infra-stack.ts

```diff
@@ -486,11 +480,37 @@ export class InfraStack extends Stack {
         }));
     }
 
-    cfnInitConfig.push(InitCommand.shellCommand('set -ex;cd opensearch-dashboards;'
-        + 'sudo -u ec2-user nohup ./bin/opensearch-dashboards > dashboard_install.log 2>&1 &', {
-      cwd: '/home/ec2-user',
-      ignoreErrors: false,
-    }));
+    // If OSD Url is present
+    if (props.dashboardsUrl !== 'undefined') {
```

Comment:
Do we need to check for empty string as well?

Reply:
Not required at the moment.